src/Microsoft.DotNet.ServiceFabric.ServiceHost/LoggedServiceRemoting/AsyncInterceptor.cs

```diff
@@ -39,6 +39,11 @@ public abstract class AsyncInterceptor : IInterceptor
         public virtual void Intercept(IInvocation invocation)
         {
             Type retType = invocation.Method.ReturnType;
+            if (retType == typeof(void))
+            {
+                throw new NotSupportedException("Void returning methods are not supported");
```

Comment:
NIT: Maybe reference AsyncInterceptor in this message. That would make the error easier to understand without having to dig through the stack trace.

Reply:
True true. :-)